scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
+    "            neighbour_bin_values = neighbour_tof[:,c]\n",
+    "\n",
+    "            # check if values are identical, e.g. all elements are zero\n",
+    "            if np.all(neighbour_bin_values == neighbour_bin_values[0]):\n",
+    "                continue\n",
+    "\n",
+    "            # sorting those values of the same time bin and return as indices of array (argsort).\n",
+    "            neighbour_ind_sort = np.argsort(neighbour_bin_values)\n",
+    "\n",
+    "            # for array with odd number of elements, sort and take centre value\n",
+    "            if len(neighbour_bin_values) % 2:\n",
+    "                median_decision[ neighbour_ind_sort[len(neighbour_ind_sort)//2] ] += 1\n",
+    "            # for array with even number of elements, sort and take both high and low centre value\n",
+    "            elif len(neighbour_bin_values) % 2 == 0:\n",
+    "                median_decision[ neighbour_ind_sort[(len(neighbour_ind_sort)-1) // 2] ] += 1 # low median\n",
+    "                median_decision[ neighbour_ind_sort[len(neighbour_ind_sort) // 2] ] += 1 # high median\n",
+    "\n",
+    "        # choosing spectrum according to median_decision list\n",
+    "        # TODO argmax returns first maximum value. In case of many maximum values make average of spectra, or find other reasonable criteria.\n",
+    "        y = neighbour_tof[np.argmax(median_decision)]\n",
+    "\n",
+    "        # TODO ws.setE. Uncertanties not delt with\n",
+    "        filtered_var[\"spectrum\", spec_num].values = y\n",
+    "\n",
+    "    return filtered_var\n",
+    "    \n",
+    "\n",
+    "for k in wavelength.keys():\n",
+    "    if k == \"reference\":  # This can be removed once #1176 is resolved\n",
+    "        continue\n",
+    "\n",
+    "    wavelength[k] = median_filter(dataset=wavelength, variable_key=k)\n",
+    "    # TODO export_to_tiff_stack for each"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Adds the component info needed for converting units\n",
-    "sc.compat.mantid.load_component_info(stitched, instrument_file)\n",
-    "stitched"
+    "# As each detector has a different relative position to the sample we\n",
+    "# need to rebin to a common set of wavelength bins for all\n",
+    "wav_x_vals = wavelength[\"sample\"].coords[\"wavelength\"].values\n",
+    "start_wave = np.max(wav_x_vals[:, 0])\n",
+    "stop_wave = np.min(wav_x_vals[:, -1])\n",
+    "bin_width_wave = np.mean(wav_x_vals[:, 1:]-wav_x_vals[:, :-1])\n",
+    "\n",
+    "wavelength_rebin_params = sc.Variable([\"wavelength\"], values=np.arange(start_wave, stop_wave, bin_width_wave,\n",
+    "                                                           dtype=np.float64))\n",
```

Comment:
`rebin_params` is a bit misleading maybe? It is actually already the bin edges (the future coord), the params (in Mantid speak) are what you pass to `np.arange`.